<?xml version="1.0"?>
<launch>
    <!--
    <arg name="start_pt" default= '-9.16926193237, -15.6781167984, 0.8978652, 0'/>
  -->
    <arg name="start_pt" default= '-4.5, -5.5, 2.47, 0'/>
    <arg name="use_rviz" default="true"/>
    <arg name="use_matplotlib" default="false"/>
    <arg name="map_file" default="$(find svea_core)/src/scripts/maps/floor2.simplemap" />
    <arg name="ctrl_input" default="mouse"/>

    <arg name="mouse_vel" default="mouse_vel"/>
    <arg name="holonomic" default="false"/>
    <arg name="plot" default="false"/>
    <arg name="latency" default="0.0"/>
    <arg name="eval_type" default="test"/>
    <arg name="is_eval" default="false"/>



    <group if="$(eval use_rviz == true)">
        <node type="rviz" name="rviz" pkg="rviz" args="-d $(find svea_core)/src/scripts/rviz/SVEA_ab_twinteleop.rviz" />
    </group>

    <node pkg="mrpt_map" type="map_server_node" name="map_server_node" output="screen">
        <param name="debug" value="false"/>
        <param name="frame_id" value="map"/>
        <param name="frequency" value="1"/>
        <param name="ini_file" value="$(find svea_sensors)/params/mrpt/pf_localization.ini"/>
        <param name="map_file" value="$(arg map_file)"/>
    </node>

    <node pkg="svea_core" type="telerobot_module.py" name="telerobot_module" output="screen">
        <param name="start_pt" value="$(arg start_pt)"/>
        <param name="use_rviz" value="$(arg use_rviz)"/>
        <param name="use_matplotlib" value="$(arg use_matplotlib)"/>
        <param name="is_eval" value="$(arg is_eval)"/>
    </node>

    <node pkg="svea_core" type="control_tower_module.py" name="control_tower_module" output="screen">
        <param name="start_pt" value="$(arg start_pt)"/>
        <param name="use_rviz" value="$(arg use_rviz)"/>
        <param name="use_matplotlib" value="$(arg use_matplotlib)"/>
        <param name="ctrl_input" value="$(arg ctrl_input)"/>
        <param name="latency" value="$(arg latency)"/>
        <param name="eval_type" value="$(arg eval_type)"/>
    </node>

    <!--path planning-->

    <node pkg="svea_core" type="genetic_path_planner.py" name="path_planner" output="screen"/>

    <!--key teleop-->

    <node pkg="key_teleop" type="key_teleop.py" name="teleop" output="screen" launch-prefix="xterm -e" if="$(eval ctrl_input == 'key')"/>


    <!--mouse teleop-->

    <node pkg="mouse_teleop" type="mouse_teleop.py" name="mouse_teleop" if="$(eval ctrl_input == 'mouse')">
        <rosparam file="$(find mouse_teleop)/config/mouse_teleop.yaml" command="load"/>
        <param name="holonomic" value="$(arg holonomic)"/>
        <remap from="mouse_vel" to="$(arg mouse_vel)"/>
    </node>

    <group if="$(arg plot)">
        <node pkg="rqt_plot" type="rqt_plot" name="rqt_plot"
        args="$(arg mouse_vel)/linear/x $(arg mouse_vel)/linear/y $(arg mouse_vel)/angular/z"/>
    </group>

</launch>
